airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -566,10 +566,10 @@ def _call_with_retry(f: Callable[[], None]) -> None:
             if gzip:
                 if isinstance(data, str):
                     data = bytes(data, encoding)
-                out = BytesIO()
-                with gz.GzipFile(fileobj=out, mode="w") as f:
-                    f.write(data)
-                data = out.getvalue()
+                with BytesIO() as out:
+                    with gz.GzipFile(fileobj=out, mode="w") as f:
+                        f.write(data)
+                    data = out.getvalue()
```

Comment:
```suggestion
                with BytesIO() as out, gz.GzipFile(fileobj=out, mode="w") as f:
                    f.write(data)
                    data = out.getvalue()
```

I believe this works.